uiowa/uiowa
# Classrooms - adding API fields

Resolves #6032
Resolves https://github.com/uiowa/uiowa/issues/6041

<details><summary>Todo</summary>

- [x] Setup rabbit hole/rabbit_hole taxonomy for the vocabularies we don't want visitors to see (access denied) (all but units)
- [x] Remove "link entity label" for room field renders (all but units)
- [x] Make sure webmasters have add/edit/delete for room_features, technology_features, units (not room types)
- [x] Decide what to do about room schedule https://github.com/uiowa/uiowa/issues/6032#issuecomment-1358441441
- [x] Add help text to disabled fields and related taxonomies
 </details>

# How to test

- Check out classrooms_api_fields and pull
- `ddev blt ds --site=classrooms.uiowa.edu`
- `ddev drush @classrooms.local uli`
- Create a room - select a building, add a room ID, and save (can get rooms from https://classrooms.uiowa.edu/)
- Several fields should now have values. Go back to Edit and open Imported fieldgroup to see what fields did/didn't pull data.
- Confirm saved field data links up to the API - https://api.maui.uiowa.edu/maui/api/pub/registrar/courses/AstraRoomData/{building_id}/{room_id}
- For the 5 taxonomy fields (Responsible Unit, Room Features, Room Type, Scheduling Region, Technology Features) you'll need to add some of the terms from the API data to the taxonomies before saving again to see the data imported. 
- Masquerade as different roles, only webmasters should be able to view/add/delete taxonomy terms. 
- Also check rabbit hole is set up correctly. - trying viewing an added taxonomy term for anything other than the units - for example https://classrooms.uiowa.ddev.site/room-types/auditorium - in an incognitio window. Page should return access denied. Can also test that a units taxonomy term is accessible in incognito.
